snuba/query/mql/mql_context.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Sequence
+
+from snuba.query.parser.exceptions import ParsingException
+
+
+@dataclass
+class MQLContext:
+    """
+    The MQL string alone is not enough to fully describe a query.
+    This class contains all of the additional information needed to
+    execute a metrics query in snuba.
+
+    This class is meant as a convenient way to validate that all of the
+    required fields are present and of the correct type. It's not meant
+    to build actual expressions. It should line up with the schema of
+    the request itself. It should not do any validation beyond type checks.
+    """
+
+    entity: str
+    start: str
+    end: str
+    rollup: Rollup
+    scope: MetricsScope
+    indexer_mappings: dict[str, str | int]
+    limit: int | None
+    offset: int | None
+
+    def __post_init__(self) -> None:
+        self.validate()
+
+    def validate(self) -> None:
+        for field in ["entity", "start", "end"]:
+            if not isinstance(getattr(self, field), str) or getattr(self, field) == "":
+                raise ParsingException(f"MQL context: {field} must be a str")
```

Comment:
FYI: none of the parsing exceptions are covered by tests.  not a blocker for me but something to consider